bigcode_eval/tasks/humaneval.py

```diff
@@ -95,11 +151,17 @@ def process_results(self, generations, references):
         :param references: list(str)
             list of str containing refrences
         """
-        results, _ = compute_code_eval(
+        results, fine_grain_results = compute_code_eval(
             references=references,
             predictions=generations,
             k=self.k,
             num_workers=self.num_workers,
             timeout=self.timeout,
         )
-        return results
+
+        simplified_fine_grain_res = {}
+        for task_id in fine_grain_results:
```

Comment:
It would be clearer to just pass the results here and do any simplification at save. That way all the logic is centralized to either inside the compute code eval or when exporting.